### PR TITLE
Gui

### PR DESCRIPTION
**Pull request by DavidLP**
This merges the latest features and fixes of the GUI into `development`. This includes proper multi-threading for analysis and plotting, re-running tabs, new progressbar, etc.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -23,7 +23,7 @@ var/
 # Usually these files are written by a python script from a template
 # before PyInstaller builds the exe, so as to inject date/other infos into it.
 *.manifest
-*.spec
+#*.spec
```
Comment (laborleben) [inline .gitignore:26]: Why?

File: .gitignore [hunk 2]
```diff
@@ -58,4 +58,5 @@ tmp*.*
 # Downloaded data files
 fixtures
 testbeam_analysis/testing/simulation
-testbeam_analysis/testing/examples
\ No newline at end of file
+testbeam_analysis/testing/examples
+testbeam_analysis/examples/data
```
Comment (laborleben) [inline .gitignore]: Add newline at EOF.

File: scripts/create_doc.sh [hunk 3]
```diff
@@ -1,11 +1,19 @@
...
+  git remote set-url --push origin "git@github.com:$TRAVIS_REPO_SLUG"
+  export ${!TRAVIS*}
+  sphinx-versioning push -r development -w master -w development -b docs gh-pages .
+else
+  echo 'Did not create docs'
+  echo $TRAVIS_BRANCH
+  echo $TRAVIS_OS_NAME
+  echo $TRAVIS_PYTHON_VERSION
+fi
+exit 0
```
Comment (laborleben) [inline scripts/create_doc.sh]: Newline.

File: testbeam_analysis/gui/analysis_widgets.py [hunk 4]
```diff
@@ -603,7 +648,7 @@ def _init_tabs(self):
                         tmp_setup[s_key] = [self.setup[s_key][i]]  # FIXME: Does not work properly without list
                     else:
                         tmp_setup[s_key] = self.setup[s_key][i]
-                elif isinstance(self.setup[s_key], int) or isinstance(self.setup[s_key], str):
+                else:  # isinstance(self.setup[s_key], int) or isinstance(self.setup[s_key], str):
```
Comment (laborleben) [inline testbeam_analysis/gui/analysis_widgets.py]: If not used anymore, remove.

File: testbeam_analysis/gui/analysis_widgets.py [hunk 5]
```diff
@@ -613,7 +658,7 @@ def _init_tabs(self):
                         tmp_options[o_key] = [self.options[o_key][i]]  # FIXME: Does not work properly without list
                     else:
                         tmp_options[o_key] = self.options[o_key][i]
-                elif isinstance(self.options[o_key], int) or isinstance(self.options[o_key], str):
+                else:  # isinstance(self.options[o_key], int) or isinstance(self.options[o_key], str):
```
Comment (laborleben) [inline testbeam_analysis/gui/analysis_widgets.py]: remove?

File: testbeam_analysis/gui/analysis_widgets.py (hunk 4 above)
Comment (laborleben) [reply testbeam_analysis/gui/analysis_widgets.py]: ...maybe a bad idea, what if if-statement is not True? Else-statement is certainly missing or comment why all other cases are not treated.

File: testbeam_analysis/gui/analysis_widgets.py (hunk 5 above)
Comment (laborleben) [reply testbeam_analysis/gui/analysis_widgets.py]: ...same as above.

File: testbeam_analysis/gui/option_widgets.py [hunk 6]
```diff
@@ -30,7 +30,7 @@ def __init__(self, name, default_value, optional, tooltip, dtype, parent=None):
         self.edit.setAlignment(QtCore.Qt.AlignCenter)
         validator = QtGui.QDoubleValidator()
         self.edit.setValidator(validator)
-        self.edit.setMaxLength(3)
+        #self.edit.setMaxLength(3)  #FIXME: No need for only 3 digits
```
Comment (laborleben) [inline testbeam_analysis/gui/option_widgets.py]: Fix or remove.

File: testbeam_analysis/gui/option_widgets.py [hunk 7]
```diff
@@ -235,7 +235,7 @@ def __init__(self, name, labels, default_value, optional, tooltip, parent=None):
             edit = QtWidgets.QLineEdit()
             edit.setAlignment(QtCore.Qt.AlignCenter)
             edit.setValidator(QtGui.QDoubleValidator())
-            edit.setMaxLength(3)
+            #edit.setMaxLength(3)  #FIXME: No need for only 3 digits
```
Comment (laborleben) [inline testbeam_analysis/gui/option_widgets.py]: Fix or remove.

**Review by laborleben — COMMENTED** [review]
Can you describe in short what have been done in this branch?

I'm the wrong reviewer for this branch anyways :-)

File: .gitignore (hunk 1 above)
Comment (DavidLP) [reply .gitignore:26]: We have an [installer](https://github.com/SiLab-Bonn/testbeam_analysis/blob/gui/scripts/pyinstaller_linux.spec) for TBA. It works under Linux after some heavy hacking... Windows fails.
Executable here: https://github.com/SiLab-Bonn/testbeam_analysis/releases/download/0.0.1/Linux_Binary.bz2.run

Comment (DavidLP) [reply .gitignore:26]: Although this needs clean up. I see my path hardcoded there ;-)

**Comment by DavidLP** [comment]
For whatever reason `pyqt` is not installed with `matplotlib` on OSX builds.
@leloup314 Can you add `conda install pyqt` in `.travis`?

**Review by leloup314 — APPROVED** [review]
I approve.

**Comment by leloup314** [comment]
I think coverage is wrong. On my latest passing commit coverage increased by 17.9% when pulling onto `development`. The issue seems to be that now the current and latest commit **within this PR** are compared by coveralls instead of comparing this PR to coverage of `development`. This issue is also stated [here](https://github.com/lemurheavy/coveralls-public/issues/1019)

**Comment by leloup314** [comment]
@DavidLP Can you merge? Or do you request any changes?

**Comment by DavidLP** [comment]
Thanks!

**Pull request merged**